File: FrontEnd/src/models/userDto.ts
import { UserRoleDTO } from "~/models/userRoleDto";
import { RoleDto } from "@models/roleDTO";

export interface UserDTO {
  id: string;
  fullname: string;
  description: string;
  password: string;
  email: string;
  phone: string;
  userTypeId: string;
  address: string;
  status: number;
  createdDate: Date;
  userCode: string;
  isLocked: boolean;
  isDeleted: boolean;
  unitId: string;
  isActive: boolean;
  createdBy: string;
  activeCode: string;
  avatar: string;
  refreshToken: string;
  roles: RoleDto[];
}

export interface UserResponse {
  data: UserDTO;
  roleList: UserRoleDTO[];
  isAdmin: boolean;
}

File: BackEnd/ExamManagement/DPD.HR.Business.Dto/Dto/UserDto.cs
namespace DPD.HumanResources.Dtos.Dto
{
    public class UserDto
    {
        public Guid Id { get; set; }
        public string? Fullname { get; set; } = string.Empty;
        public string? Description { get; set; }
        public string Email { get; set; } = string.Empty;
        public string? Phone { get; set; }
        public Guid UserTypeId { get; set; }
        public string? Address { get; set; }
        public int? Status { get; set; }
        public DateTime? CreatedDate { get; set; }
        public Guid? CreatedBy { get; set; }
        public string? UserCode { get; set; }
        public bool IsLocked { get; set; }
        public bool IsDeleted { get; set; }
        public bool IsActive { get; set; }
        public Guid UnitId { get; set; }
        public string ActiveCode { get; set; } = string.Empty;
        public string? Avatar { get; set; }
    }
}
